### PR TITLE
Fix[MQB]: respect protocol file size limits

### PR DESCRIPTION
**Pull request by 678098**
Before this PR, it was possible to set a file size not complaint to the protocol limitations. In this case, the cluster will work until it gets the first offset overflow, and then crash.

This PR:
- Adds precise pre-calculated constants for DATA, JOURNAL, QLIST file sizes hard limits.
- Adds the 1st line of checks in `mqbblp::StorageManager::start` (for non-FSM) and `mqbc::StorageManager::start` (for FSM) to gracefully return a failure RC on file size overflow.
- Adds the 2nd line of checks (asserts in `FileStoreSet::setDataFileSize` etc), they should be never triggered in a normal workflow but left here just to be sure.
- Improves consistency of RCs returned from [`mqbblp`/`mqbc`]`::StorageManager::start`
- Adds UT to check file size hard limits for `mqbc::StorageManager` (there are no UTs for `mqbblp::StorageManager`).

The error logs look like this:
```
06DEC2024_00:29:20.649 25904:8394919744 ERROR /blazingmq/src/groups/mqb/mqbc/mqbc_storagemanager.cpp:3485 MQBC.STORAGEMANAGER Configured maxDataFileSize (34359738361) exceeds the protocol limit (34359738360) 

06DEC2024_00:29:20.649 25904:8394919744 ERROR /blazingmq/src/groups/mqb/mqbc/mqbc_storagemanager.cpp:3495 MQBC.STORAGEMANAGER Configured maxJournalFileSize (17179869181) exceeds the protocol limit (17179869180) 

06DEC2024_00:29:20.649 25904:8394919744 ERROR /blazingmq/src/groups/mqb/mqbc/mqbc_storagemanager.cpp:3505 MQBC.STORAGEMANAGER Configured maxQlistFileSize (17179869181) exceeds the protocol limit (17179869180) 

06DEC2024_00:29:20.649 25904:8394919744 ERROR /blazingmq/src/groups/mqb/mqbc/mqbc_storagemanager.cpp:3485 MQBC.STORAGEMANAGER Configured maxDataFileSize (8796093020160) exceeds the protocol limit (34359738360) 
```

### CODE REVIEW

File: src/groups/mqb/mqbs/mqbs_filestoreprotocol.h [hunk 1]
```diff
@@ -59,12 +59,15 @@
...
-//                               (courtesy MessageRecord.d_messageOffsetDwords)
-//       Max Journal file size..........................: 16GB
-//                          (courtesy bmqp::StorageHeader.d_journalOffsetWords)
-//       Max Qlist file size............................: 16GB
-//                         (courtesy QueueOpRecord.d_queueUriRecordOffsetWords)
+//       Max Data file size.............................: 34359738360 bytes
+//                                                        or 32GiB - 8B
+//               (See also: mqbs::FileStoreProtocol::k_MAX_DATA_FILE_SIZE_HARD)
+//       Max Journal file size..........................: 17179869180 bytes
+//                                                        or 16GiB - 8B
```
Comment (678098) [inline src/groups/mqb/mqbs/mqbs_filestoreprotocol.h:66]:
```suggestion
//                                                        or 16GiB - 4B
```

File: src/groups/mqb/mqbs/mqbs_filestoreprotocol.h [hunk 2]
```diff
@@ -59,12 +59,15 @@
...
+//            (See also: mqbs::FileStoreProtocol::k_MAX_JOURNAL_FILE_SIZE_HARD)
+//       Max Qlist file size............................: 17179869180 bytes
+//                                                        or 16GiB - 8B
```
Comment (678098) [inline src/groups/mqb/mqbs/mqbs_filestoreprotocol.h:69]:
```suggestion
//                                                        or 16GiB - 4B
```

File: src/groups/mqb/mqbc/mqbc_storagemanager.t.cpp [hunk 3]
```diff
@@ -2892,6 +2892,109 @@ static void test18_primaryHealingStage1SelfHighestSendsDataChunks()
...
+                mockOnPartitionPrimaryStatus,
+                bmqtst::TestHelperUtil::allocator());
+
+            bmqu::MemOutStream errorDescription(
+                bmqtst::TestHelperUtil::allocator());
+            const int rc = storageManager.start(errorDescription);
+            if (rc == 0) {
+                storageManager.stop();
+            }
+            ASSERT_EQ_D("line: " << line << ", expected failure: "
```
Comment (kaikulimu) [inline src/groups/mqb/mqbc/mqbc_storagemanager.t.cpp:2950]:
```suggestion
            BMQTST_ASSERT_EQ_D("line: " << line << ", expected failure: "
```

**Review by kaikulimu — CHANGES_REQUESTED** [review]
One minor comment

File: src/groups/mqb/mqbc/mqbc_storagemanager.t.cpp (hunk 3 above)
Comment (kaikulimu) [reply src/groups/mqb/mqbc/mqbc_storagemanager.t.cpp:2950]: Might to update with base branch before this can compile

**Review by kaikulimu — APPROVED** [review]

**Pull request merged**